docs/introduction.rst

```diff
@@ -304,4 +304,60 @@ Cons
 
 Saliency Algorithms
 --------------------------------
-*Discuss the provided xaitk-saliency algorithms in terms of the above.*
+.. Discuss the provided xaitk-saliency algorithms in terms of the above.
+
+The xaitk-saliency API consists of a number of object-oriented interfaces for saliency heatmap generation, which
+focus on black-box visual saliency. Three types of implementations of these interfaces are provided: Image
+Perturbation, Heatmap Generation, and End-to-End Saliency Generation.
+
+
+Image Perturbation
+^^^^^^^^^^^^^^^^^^^
+These implementations are occlusion-based saliency algorithms that perform perturbations on image pixels.
+
+* RISEGrid - Transforms an input reference image into a number of mask matrices indicating the perturbed regions and
+  generates an importance map indicating how salient each pixel is for the model's prediction.
+  See the "DRISE" Jupyter Notebook example.
+
+* SlidingRadial - Produces perturbation matrices generated by sliding a radial occlusion area with configured radius
+  over the area of an image. When the two radius values are the same, circular masks are generated; otherwise,
+  elliptical masks are generated. See the "Radial_Image_Perturbation" and "atari_deepRL_saliency"
+  Jupyter Notebook examples.
+
+* SlidingWindow - Produces perturbation matrices based on hard, block-y occlusion areas as generated by sliding a
+  window of a configured sizse over the area of an image. See the "SimilarityScoring" Jupyter Notebook example.
+
+
+Heatmap Generation
+^^^^^^^^^^^^^^^^^^^
+These implementations transform black-box image classification scores into saliency heatmaps.
+
+* DRISEScoring - Transforms black-box object detector predictions into visual saliency heatmaps.
+  See the "DRISE" Jupyter Notebook example.
+
+* OcculsionScoring - Transforms black-box image classification scores into saliency heatmaps.
+  This should require a sequence of per-class confidences predicted on the reference image, a number of per-class
+  confidences as predicted on perturbed images, as well as the masks of the reference image perturbations.
+  See the "SuperPixelSaliency" Jupyter Notebook example.
```

Comment:
Both `OcclusionScoring` here and `RISEScoring` are used internally in the stacked implementations below, although somewhat hidden from the user in the `OcclusionSaliency` notebook. It might still be worth mentioning here though.

Reply:
I added this note to each bullet (highlighted in bold) is that what you were intending?

* OcclusionScoring - Transforms black-box image classification scores into saliency heatmaps.
  This should require a sequence of per-class confidences predicted on the reference image, a number of per-class
  confidences as predicted on perturbed images, as well as the masks of the reference image perturbations.
  See the "SuperPixelSaliency" Jupyter Notebook example **(also used internally in the OcclusionSaliency notebook)**.

* RISEScoring - Generates saliency maps that are based on the original RISE implementation,
  utilizing only the input perturbed image confidence predictions and not the reference image confidences
  **(also used internally in the OcclusionSaliency notebook)**.